### PR TITLE
Add chainguard docker image

### PR DESCRIPTION
**Pull request by breskeby**
This added Wolfi images based on Chainguard-base to our docker images. For Chainguard we resolve tini from the Wolfi registry instead of resolving it from GitHub.

This PR doesn't handle any optimisation to how we setup our docker images. 
But we should probably have follow up PRs addressing making them better to maintain.

We could also improve the layering of the docker images to have the changing bits in a later layer (e.g. the elasticsearch bistro that _ALWAYS_ fails) further up in the layer so we can reuse more layers when rebuilding.

### CODE REVIEW

**Review by mark-vieira — COMMENTED** [review]
I think the only thing missing here is adding this to our existing packaging tests. Looks like we'll need to [add it here](https://github.com/elastic/elasticsearch/blob/a6f2a4df8b80abf65f905c34438731813d0c21ac/build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/distribution/InternalElasticsearchDistributionTypes.java#L24).

**Comment by breskeby** [comment]
> I think the only thing missing here is adding this to our existing packaging tests. Looks like we'll need to [add it here](https://github.com/elastic/elasticsearch/blob/a6f2a4df8b80abf65f905c34438731813d0c21ac/build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/distribution/InternalElasticsearchDistributionTypes.java#L24).

Added this and triggered a manual run here: https://buildkite.com/elastic/elasticsearch-periodic-packaging/builds/3560

**Comment by mark-vieira** [comment]
Packaging tests look good.

**Review by mark-vieira — APPROVED** [review]

File: distribution/docker/src/docker/Dockerfile [hunk 1]
```diff
@@ -152,6 +157,15 @@ RUN ${package_manager} update --setopt=tsflags=nodocs -y && \\
       nc shadow-utils zip findutils unzip procps-ng && \\
     ${package_manager} clean all
 
+<% } else if (docker_base == "wolfi") { %>
+RUN <%= retry.loop(package_manager,
+          "export DEBIAN_FRONTEND=noninteractive && \n" +
+          "      ${package_manager} update && \n" +
+          "      ${package_manager} upgrade && \n" +
+          "      ${package_manager} add --no-cache \n" +
+          "        bash ca-certificates curl libsystemd netcat-openbsd p11-kit p11-kit-trust shadow tini unzip zip zstd && \n" +
```
Comment (mgreau) [inline distribution/docker/src/docker/Dockerfile:166]: @breskeby you should be able to remove `ca-certificates` as the `chainguard-base` image provides it:
```
docker run --entrypoint=cat docker.elastic.co/wolfi/chainguard-base:latest@sha256:c16d3ad6cebf387e8dd2ad769f54320c4819fbbaa21e729fad087c7ae223b4d0 /etc/apk/world
apk-tools=2.14.4-r0
busybox=1.36.1-r10
ca-certificates-bundle=20240705-r0
chainguard-baselayout=20230214-r10
glibc-locale-posix=2.40-r1
glibc=2.40-r1
ld-linux=2.40-r1
libcrypt1=2.40-r1
libcrypto3=3.3.1-r5
libssl3=3.3.1-r5
libxcrypt=4.4.36-r7
wolfi-base=1-r6
wolfi-baselayout=20230201-r15
wolfi-keys=1-r8
zlib=1.3.1-r4
```

**Pull request merged**